src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,169 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+        size = 0;
+        head = null;
+        tail = null;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (size == 0) {
+            head = node;
+            tail = node;
+        } else {
+            node.prev = tail;
+            tail = node;
+            node.prev.next = tail;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkSizeAdd(index);
+        if (index == 0 && size == 0 || index > 0 && index == size) {
+            add(value);
+        } else {
+            link(value, index);
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return find(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        T oldValue = find(index).value;
+        find(index).value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T oldValue = find(index).value;
+        unlink(find(index));
+        size--;
+        return oldValue;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> newNode = new Node<>(null, object, null);
+        for (int i = 0; i < size; i++) {
+            if (newNode.value == find(i).value
+                    || newNode.value != null && newNode.value.equals(find(i).value)) {
+                unlink(find(i));
+                size--;
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndex(int index) {
+        if (!(index >= 0 && index < size)) {
+            throw new IndexOutOfBoundsException("Index out of bounds " + index);
+        }
+    }
+
+    private void checkSizeAdd(int index) {
+        if (!(index >= 0 && index <= size)) {
+            throw new IndexOutOfBoundsException("Can't add element to index " + index);
+        }
+    }
+
+    private Node<T> find(int index) {
+        Node<T> node = head;
+        if (index == 0) {
+            return node;
+        }
+        if (index == size - 1) {
+            return tail;
+        }
```

Comment:
I don't think this check is necessary, if you allocate two separate functions for looping from the beginning and end, you just won't enter the loop, but will return head or tail at once, so you can reduce your code.